### PR TITLE
V2X: Update os interface

### PR DESCRIPTION
**Pull request by QIU1995NONAME**

### CODE REVIEW

File: modules/v2x/v2x_proxy/os_interface/os_interface.cc [hunk 1]
```diff
@@ -71,8 +71,11 @@ bool OsInterFace::InitWriters() {
   v2x_traffic_light_hmi_writer_ =
       node_->CreateWriter<::apollo::perception::TrafficLightDetection>(
           FLAGS_v2x_traffic_light_for_hmi_topic);
-
-  return !!v2x_obu_traffic_light_writer_ && !!v2x_traffic_light_writer_;
+  v2x_obstacles_internal_writer_ =
+      node_->CreateWriter<apollo::v2x::V2XObstacles>(
+          FLAGS_v2x_internal_obstacle_topic);
+  return !!v2x_obu_traffic_light_writer_ && !!v2x_traffic_light_hmi_writer_ &&
+         !!v2x_obstacles_internal_writer_ && !!v2x_traffic_light_writer_;
```
Comment (lfcarol) [inline modules/v2x/v2x_proxy/os_interface/os_interface.cc:78]: What's meaning '!!' ?

File: modules/v2x/v2x_proxy/os_interface/BUILD [hunk 2]
```diff
@@ -17,6 +17,7 @@ cc_library(
         "//modules/v2x/proto:v2x_obu_rsi_cc_proto",
         "//modules/v2x/proto:v2x_obu_traffic_light_cc_proto",
         "//modules/v2x/proto:v2x_traffic_light_cc_proto",
+        "//modules/v2x/proto:v2x_obstacles_cc_proto",
```
Comment (changsh726) [inline modules/v2x/v2x_proxy/os_interface/BUILD:20]: Notice the order, you can fix this by `./apollo.sh format -b modules/v2x/v2x_proxy/os_interface`

File: modules/v2x/v2x_proxy/os_interface/os_interface.cc (hunk 1 above)
Comment (QIU1995NONAME) [reply modules/v2x/v2x_proxy/os_interface/os_interface.cc:78]: Convert pointer to bollean instead of use lots of ```nullptr !=```

Comment (lfcarol) [reply modules/v2x/v2x_proxy/os_interface/os_interface.cc:78]: Please removed '!!' or replaced it by 'nullptr != XXX' .

Comment (QIU1995NONAME) [reply modules/v2x/v2x_proxy/os_interface/os_interface.cc]: done

File: modules/v2x/v2x_proxy/os_interface/os_interface.h [hunk 3]
```diff
@@ -60,6 +61,9 @@ class OsInterFace {
   void SendV2xTrafficLightToOs(
       const std::shared_ptr<::apollo::v2x::IntersectionTrafficLightData> &msg);
 
+  void SendV2xInternalObstacles2Sys(
+      const std::shared_ptr<apollo::v2x::V2XObstacles> &msg);
```
Comment (lfcarol) [inline modules/v2x/v2x_proxy/os_interface/os_interface.h:65]: Please keep 'apollo::v2x::V2XObstacles' style consistent with others.

File: modules/v2x/v2x_proxy/os_interface/os_interface.cc [hunk 4]
```diff
@@ -58,7 +58,7 @@ bool OsInterFace::InitReaders() {
         }
         cond_planning_.notify_one();
       });
-  return !!localization_reader_ && !!planning_reader_;
+  return nullptr != localization_reader_ && nullptr != planning_reader_;
```
Comment (changsh726) [inline modules/v2x/v2x_proxy/os_interface/os_interface.cc:61]: Why not `return localization_reader_ && planning_reader_;`

Comment (QIU1995NONAME) [reply modules/v2x/v2x_proxy/os_interface/os_interface.cc:61]: To emphasize that these are two pointers, not two boolean values. ```&&``` should **ONLY** be used to connect boolean values

**Review by changsh726 — APPROVED** [review]

**Pull request merged**